crates/nu-command/src/network/url/parse.rs

```diff
@@ -0,0 +1,178 @@
+        "Parses a url"
+    }
+
+    fn search_terms(&self) -> Vec<&str> {
+        vec!["scheme", "hostname", "port", "path", "parameters"]
+    }
+
+    fn run(
+        &self,
+        engine_state: &EngineState,
+        _: &mut Stack,
+        call: &Call,
+        input: PipelineData,
+    ) -> Result<PipelineData, ShellError> {
+        parse(input.into_value(call.head), engine_state)
+    }
+
+    fn examples(&self) -> Vec<Example> {
+        vec![Example {
+            description: "Parses a url",
+            example: "'http://www.example.com:8081/foo/bar?p=section&s=&f[name]=vldc' | url parse",
+            result: Some(Value::Record {
+                cols: vec![
+                    "scheme".to_string(),
+                    "host".to_string(),
+                    "port".to_string(),
+                    "path".to_string(),
+                    "query".to_string(),
+                    "params".to_string(),
+                ],
+                vals: vec![
+                    Value::test_string("http"),
+                    Value::test_string("www.example.com"),
+                    Value::test_string("8081"),
+                    Value::test_string("/foo/bar"),
+                    Value::test_string("butter"),
+                    Value::Record {
+                        cols: vec!["p".to_string(), "s".to_string(), "f[name]".to_string()],
+                        vals: vec![
+                            Value::test_string("section"),
+                            Value::test_string(""),
+                            Value::test_string("vldc"),
+                        ],
+                        span: Span::test_data(),
+                    },
+                ],
+                span: Span::test_data(),
+            }),
+        }]
+    }
+}
+
+fn get_url_string(value: &Value, engine_state: &EngineState) -> String {
+    value.into_string("", engine_state.get_config())
+}
+
+fn parse(value: Value, engine_state: &EngineState) -> Result<PipelineData, ShellError> {
+    let url_string = get_url_string(&value, engine_state);
+
+    let result_url = Url::parse(url_string.as_str());
+
+    let head = value.span()?;
+
+    match result_url {
+        Ok(url) => {
+            let cols = vec![
+                String::from("scheme"),
+                String::from("host"),
+                String::from("port"),
+                String::from("path"),
+                String::from("query"),
+                String::from("params"),
+            ];
+            let mut vals: Vec<Value> = vec![
+                Value::String {
+                    val: String::from(url.scheme()),
+                    span: head,
+                },
+                Value::String {
+                    val: String::from(url.host_str().unwrap_or("")),
+                    span: head,
+                },
+                Value::String {
+                    val: url
+                        .port()
+                        .map(|p| p.to_string())
```

Comment:
I believe we should leave this as an integer seeing as Rust is providing it as an int.

Reply:
in that case, what should be the output when there's no `port` on `url` ? An empty string? I could do something like this for `port` value:
```
match url.port() {
    Some(p) => Value::Int {
        val: p as i64,
        span: head,
    },
    None => Value::String {
        val: "".into(),
        span: head,
    }
},
```
But this way, the `url parse` output type for `port` will be `string` or `int` depending on input.

Reply:
Oh you're right, sorry, I think what I said doesn't make sense until/unless Nushell gains an Option type in the future. As you have it is the way for now.